apps/application-system/api/src/app/modules/application/e2e/applicationTemplateValidation.spec.ts

```diff
@@ -0,0 +1,186 @@
+    const mockUser = {
+      nationalId: '1234567891',
+      scope: [],
+      authorization: faker.random.word(),
+      client: faker.random.word(),
+    }
+
+    const envBefore = environment.production
+    environment.production = true
+    const envNameBefore = environment.name
+    environment.name = 'production'
+
+    const results = await applicationValidationService.isTemplateReady(
+      mockUser,
+      {
+        readyForProduction: false,
+        featureFlag: ApplicationFeatures.exampleApplication,
+      },
+    )
+    expect(results).toBe(true)
+    environment.production = envBefore
+    environment.name = envNameBefore
+  })
+
+  it('Should be ready when the supplied featureFlag is disabled and readyForProd is false and the users nationalId is whitelisted on prod', async () => {
+    const mockUser = {
+      nationalId: '1234567890',
+      scope: [],
+      authorization: faker.random.word(),
+      client: faker.random.word(),
+    }
+
+    const envBefore = environment.production
+    environment.production = true
+    const envNameBefore = environment.name
+    environment.name = 'production'
+
+    const results = await applicationValidationService.isTemplateReady(
+      mockUser,
+      {
+        readyForProduction: false,
+        featureFlag: ApplicationFeatures.accidentNotification,
+      },
+    )
+    expect(results).toBe(true)
+    environment.production = envBefore
+    environment.name = envNameBefore
+  })
+
+  it('Should be ready when no featureFlag is supplied but readyForProduction is true on prod', async () => {
+    const mockUser = {
+      nationalId: '1234567890',
+      scope: [],
+      authorization: faker.random.word(),
+      client: faker.random.word(),
+    }
+
+    const envBefore = environment.production
+    environment.production = true
+    const envNameBefore = environment.name
+    environment.name = 'production'
+
+    const results = await applicationValidationService.isTemplateReady(
+      mockUser,
+      {
+        readyForProduction: true,
+      },
+    )
+    expect(results).toBe(true)
+    environment.production = envBefore
+    environment.name = envNameBefore
+  })
+
+  it('Should not be ready when no featureFlag is supplied and readyForProduction is false on prod', async () => {
+    const mockUser = {
+      nationalId: '1234567890',
+      scope: [],
+      authorization: faker.random.word(),
+      client: faker.random.word(),
+    }
+
+    const envBefore = environment.production
+    environment.production = true
+    const envNameBefore = environment.name
+    environment.name = 'production'
+
+    const results = await applicationValidationService.isTemplateReady(
+      mockUser,
+      {
+        readyForProduction: false,
+      },
+    )
+    expect(results).toBe(false)
+    environment.production = envBefore
+    environment.name = envNameBefore
+  })
+
+  it('Should be ready regardless when not running on production', async () => {
```

Comment:
We should respect the feature flag if it's set. We may not want some feature to be available on staging, or only available for some users.